### PR TITLE
NIO-1086, ensuring an error is raised, and consequently getting other than a 200 return status when operation does not complete fully

### PR DESCRIPTION
**Pull request by f1401martin**

### CODE REVIEW

File: tests/test_manager.py [hunk 1]
```diff
@@ -321,7 +329,7 @@ def test_update_all(self):
         results = project_manager.update_block([])
 
         # assert that all blocks in project_path were updated
-        self.assertEqual(len(results), 3)
+        self.assertEqual(len(results), 4)
```
Comment (mattdodge) [inline tests/test_manager.py]: Why did this go to 4? Only thing I can think is that now the results have a status that is not OK in there. If so, we should check that

Comment (f1401martin) [reply tests/test_manager.py]: now as part of the operation result, a new entry is added specifying if there were any blocks requested for update that were not installed ({"not installed": [list of blocks requested but not installed}.)

Comment (f1401martin) [reply tests/test_manager.py]: One thought here would be to actually attempt to 'clone' the block if it is called for an 'update' and it is not currently in the system, thoughts?

Comment (mattdodge) [reply tests/test_manager.py]: Ok, let's add an assertion for that 4th result then. Right now this assertion doesn't align with the comment above it or the assertions below it.

**Pull request merged**